src/widgets/TooltipWidget.hpp

```diff
@@ -29,6 +29,11 @@ class TooltipWidget : public BaseWindow
     Q_OBJECT
 
 public:
+    /// @param parent The parent of this widget.
+    ///               If this widget is a window, use tooltipParentFor().
```

Comment:
Is this more accurate?

```suggestion
    ///               If the parent is a window, use tooltipParentFor().
```

Reply:
No. If we take `ChannelView` for example: `ChannelView` is not a window, but `Tooltip` is used as a window there. See [`QWidget::isWindow`](https://doc.qt.io/qt-6/qwidget.html#isWindow).

Reply:
In what scenario would we not want to use `tooltipParentFor` then?
Right now we use it both times the tooltip widget is created (`ChannelView` & `SplitHeader`)

Reply:
If the tooltip widget is embedded into another widget (or through multiple layers) that is then shown as a window.

Reply:
Since that's not a scenario that we run into now, let's just simplify this & expand to support those use cases as they come up.
Either just `#ifdef` in the constructor itself, or make `tooltipParentFor` a static function in the source file & use it in the constructor